src/kademlia/node.cpp

```diff
@@ -623,6 +610,42 @@ time_duration node::connection_timeout()
 		}
 	}
 
+	int lifetime = m_settings.item_lifetime;
+	if (0 == lifetime)
+	{
+		return d;
+	}
+
+	// item lifetime must >= 120 minutes.
+	if (lifetime < 120) lifetime = 120;
+
+	for (dht_immutable_table_t::iterator i = m_immutable_table.begin();
```

Comment:
this change conflict with https://github.com/arvidn/libtorrent/pull/123 , which probably will make it in first. Do you have any thoughts on how to resolve it? perhaps the storage should have access to the dht_settings objects.